docs/troubleshooting.md

```diff
@@ -1,79 +1,165 @@
+Usually this will provide some information about troubleshooting.
+
+For example:
+- `lsp-mode` has a [troubleshooting page](https://emacs-lsp.github.io/lsp-mode/page/troubleshooting/)
+- The VSCode Haskell extension has a [troubleshooting section](https://github.com/haskell/vscode-haskell#investigating-and-reporting-problems)
 
 Many clients provide diagnostic information about a LSP session.
 In particular, look for a way to get the status of the server, the server stderr, or a log of the messages that the client has sent to the server.
-For example, `lsp-mode` provides all of these (see its [troubleshooting page](https://emacs-lsp.github.io/lsp-mode/page/troubleshooting/) for details).
-For vscode you can read how to access the lsp session log [here])https://github.com/haskell/vscode-haskell#investigating-and-reporting-problems).
+
+## Common issues
+
+### Wrong server binary being used
+
+HLS needs to be compiled against the same version of GHC as is used in the project.
+Normally, we ship binaries for multiple versions and `haskell-language-server-wrapper` selects the correct one.
+
+If you see an error about HLS being compiled with the wrong version of GHC, then you either need to install the correct one (if you installed it yourself), or there is something going wrong with the wrapper selecting the right HLS binary to launch.
+
+### Unsupported GHC version
+
+HLS does not support every GHC version - there are a lot of them!
+Please see the [supported versions page](./supported-versions.md) for more information.
+
+In particular, support for GHC 9.0 and 9.2 is only partial; some features are unavailable with GHC >= 9.0.1, and HLS is not available at all for GHC 9.2.
+
+### Missing server or build tools
 
 The most common client-related problem is the client simply not finding the server executable or the tools needed to load Haskell code (`ghc`, `cabal`, or `stack`). So make sure that you have the right `PATH` and you have configured the client to look for the right executables.
+
+Usually this will be visible in the client's log.
+
+### Compilation failures
+
+Sometimes HLS will simply fail to do anything with a file, or give nonsensical error messages.
+The most common cause of this is that HLS is using the wrong `hie-bios` cradle to decide how to build the project (e.g., trying to use `stack` instead of `cabal`).
+The server log will show which cradle is being chosen.
+
+Using an explicit `hie.yaml` to configure the cradle can resolve the problem, see the [configuration page](./configuration.md#configuring-your-project-build).
+
+### Problems with dynamic linking
+
+The prebuilt binaries which we provide are statically linked, which means that they don't work well with projects which use dynamic linking.
+A typical symptom is the presence of errors containing `unknown symbol`, and it is common for users Arch Linux, which ships a dynamically linked version of `ghc`.
```

Comment:
```suggestion
The prebuilt binaries which we provide are statically linked, which means that they don't work well with projects which use dynamic linking.
```
What is the source for this?

The problems with statically linked HLS binaries are:

1. If `glibc` is statically linked, as in our Linux binaries, all sorts of unspecified behaviour can arise, including segfaults, when dynamically loading dependencies while compiling TH.
2. Because the GHC linker is used to dynamically load dependencies, and the GHC linker has bugs (mainly the Mach linker used in Mac OS, but also the ELF linker), HLS can segfault in projects with TH. 

2 means that HLS will crash more than `ghci`, as the latter uses the system linker instead of the GHC linker. Dynamically linked HLS binaries avoid this.

Reply:
The source is that it was already in the doc :) I don't understand the issue here so I'm happy to take a more accurate rewrite!